src/test/java/org/apache/commons/csv/CSVParserTest.java

```diff
@@ -1151,6 +1166,14 @@ public void testTrim() throws Exception {
         assertEquals("3", record.get("Z"));
         Assert.assertEquals(3, record.size());
     }
+    
+    @Test
+    public void testRepeatedHeadersAreReturnedInCSVRecordHeaderNames() throws IOException {
+        final Reader in = new StringReader("header1,header2,header1\n1,2,3\n4,5,6");
+        final Iterator<CSVRecord> records = CSVFormat.DEFAULT.withFirstRecordAsHeader().withTrim().parse(in).iterator();
+        final CSVRecord record = records.next();
+        assertEquals(Arrays.asList("header1", "header2", "header1"), record.getParser().getHeaderNames());
+    }
```

Comment:
Thank you for adding tests. Are we sure all cases are covered. You can build the site locally and look at he code coverage report.

Reply:
Yes, when I added the tests I made sure that coverage was 100% of `createHeaders`. That's no guarantee that all cases are covered of course but I think we are ok.